src/main/java/com/bookstore/service/cart/ShoppingCartServiceImpl.java

```diff
@@ -63,8 +61,7 @@ public ShoppingCartResponseDto saveItem(CartItemRequestDto requestDto) {
     @Override
     public ShoppingCartResponseDto update(Long id, CartItemRequestDtoWithoutBookId requestDto) {
         User user = (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
-        ShoppingCart cart = shoppingCartRepository.findById(user.getId())
-                .orElseThrow(() -> new EntityNotFoundException("Can't find shopping cart by id"));
+        ShoppingCart cart = getById(user.getId());
         CartItem cartItem = cart.getCartItems().stream()
                 .filter(c -> Objects.equals(c.getId(), id))
                 .findFirst()
```

Comment:
I think you can use default method from JPA repository to get cart item. Also, this code used twice, make it as a separate method and reuse it.